app/vmalert/rule/group.go

```diff
@@ -194,12 +192,16 @@ func (g *Group) newRule(qb datasource.QuerierBuilder, r config.Rule) Rule {
 	return NewRecordingRule(qb, g, r)
 }
 
-// ID return unique group ID that consists of
-// rules file and group Name
-func (g *Group) ID() uint64 {
+// GetCheckSum returns group checksum
+func (g *Group) GetCheckSum() string {
 	g.mu.RLock()
```

Comment:
I think, it's better to keep mutex there to prevent possible future data races. Even if `ID` cannot be called concurrently now, it could be changed in future and it's hard to track it.

 It could be changed, if `mu` will have an explicit list of fields that it protects.

Reply:
`ID()` doesn't require a lock because any change in the fields below will result in a new `ID()` and a new group, it should never change.

Reply:
>In this case, it's better to compute it statically as ID field and remove ID() method.

Make sense, please check https://github.com/VictoriaMetrics/VictoriaMetrics/pull/8540/commits/8f4dfe3b97dc2563498cba073e2357ab6b4286ad